proof/refine/ARM/Ipc_R.thy

```diff
@@ -5881,6 +5919,312 @@ lemma si_blk_makes_runnable':
+                               sts_invs_minor2)
+                 apply (rule_tac Q="\<lambda>_ s.
+                           st_tcb_at (\<lambda>st'. tcb_st_refs_of st' = {} \<and> \<not> awaiting_reply st') recvr s \<and>
+                           invs s \<and> valid_list s \<and> scheduler_act_not recvr s \<and>
+                           ex_nonz_cap_to recvr s \<and>
+                           recvr \<noteq> idle_thread s \<and>
+                           fault_tcb_at ((=) None) recvr s \<and>
+                           valid_release_q s \<and>
+                           active_sc_valid_refills s \<and>
+                           heap_refs_inv (sc_tcbs_of s) (tcb_scps_of s) \<and>
+                           (pred_map_eq None (tcb_scps_of s) recvr \<or> active_sc_tcb_at recvr s) \<and>
+                           not_queued recvr s \<and> not_in_release_q recvr s"
+                        in hoare_strengthen_post[rotated])
+                  apply (clarsimp split: if_split)
+                 apply (wpsimp wp: thread_set_no_change_tcb_state thread_set_cap_to
+                                   thread_set_no_change_tcb_state
+                                   thread_set_pred_tcb_at_sets_true simp: ran_tcb_cap_cases)
+                apply simp
+                apply (wpsimp wp: hoare_drop_imp)
+               apply wpsimp
+              apply wpsimp
+             apply wpsimp
+            apply (rule_tac Q="\<lambda>_. tcb_at' recvr and invs'" in hoare_strengthen_post[rotated])
+             apply (clarsimp simp: tcb_at'_ex_eq_all invs'_def valid_state'_def valid_pspace'_def)
+             apply (frule (1) tcb_ko_at_valid_objs_valid_tcb', clarsimp simp: valid_tcb'_def)
+            apply (wpsimp wp: setThreadState_Running_invs' sts_invs')
+                apply (rule_tac Q="\<lambda>_. invs' and sch_act_not recvr and ex_nonz_cap_to' recvr and tcb_at' recvr"
+                       in hoare_strengthen_post[rotated])
+                 apply (fastforce dest: global'_no_ex_cap simp: invs'_def valid_state'_def split: if_split)
+                apply (wpsimp wp: threadSet_fault_invs')
+               apply wpsimp+
+           apply (wpsimp wp: thread_get_wp')
+          apply (wpsimp wp: threadGet_wp)
+         apply (rule_tac Q="\<lambda>_. tcb_at sender and tcb_at recvr and invs and valid_list and
+                   valid_sched and scheduler_act_not recvr and not_in_release_q recvr and
+                   active_if_bound_sc_tcb_at recvr and st_tcb_at (\<lambda>a. tcb_st_refs_of a = {} \<and> \<not> awaiting_reply a) recvr and
+                   ex_nonz_cap_to recvr and not_queued recvr"
+                in hoare_strengthen_post[rotated])
+          apply (clarsimp split: option.split simp: valid_sched_def invs_def valid_state_def valid_pspace_def
+                                                   pred_tcb_at_def obj_at_def
+                           dest!: idle_no_ex_cap)
```

Comment:
nitpick: this looks a bit non-standard. `simp:` should align with `split:` and `dest!:`, I suppose?